internal/kptfile/pkgfile.go

```diff
@@ -175,6 +191,33 @@ func shouldSkipCopy(updatedDef, localDef, originalDef *yaml.MapNode, key string)
 	return localValStr == originalValStr
 }
 
+// shouldRemoveValue decides if a node with key should be removed from Def
+func shouldRemoveValue (updatedDef, localDef, originalDef *yaml.MapNode, key string) bool  {
+	localVal := localDef.Value.Field(key)
+	originalVal := originalDef.Value.Field(key)
+	updatedVal := updatedDef.Value.Field(key)
+
+	if originalVal == nil || updatedVal == nil {
+		return false
+	}
+
+	originalValStr, err := originalDef.Value.Field(key).Value.String()
```

Comment:
you can replace originalDef.Value.Field(key) with originalVal?

Reply:
Done.